platform/sampler/src/org/netbeans/modules/sampler/Sampler.java

```diff
@@ -227,7 +226,6 @@ private synchronized void stopSampling(boolean cancel, DataOutputStream dos) {
             if (cancel || samples < 1) {
                 return;
             }
-            if (SwingUtilities.isEventDispatchThread()) throw new IllegalStateException("sampling cannot be stopped from EDT");  //NOI18N
```

Comment:
consider `!GraphicsEnvironment.isHeadless() && SwingUtilities.isEventDispatchThread()`

it wouldn't surprise me if this has been already extracted into a utility somewhere since this looks to me like a common check.

Reply:
There are also other Swing usages that should be treated in a similar way

```bash
$ rg 'isDispatchThread' .
./src/main/java/org/netbeans/modules/sampler/InternalSampler.java
174:        if (EventQueue.isDispatchThread()) {
183:        if (EventQueue.isDispatchThread()) {
192:        if (EventQueue.isDispatchThread()) {
```

Reply:
Since https://github.com/apache/netbeans/pull/4052 the `InternalSampler`constructor shall throw `LinkageError` and rather `StandaloneSampler` shall be used in a non-NetBeans environment. `InternalSampler` is intended only for usage in NetBeans/Swing application.